java/core/src/java/org/apache/orc/OrcConf.java

```diff
@@ -82,6 +82,12 @@
       "If ORC reader encounters corrupt data, this value will be used to\n" +
           "determine whether to skip the corrupt data or throw exception.\n" +
           "The default behavior is to throw exception."),
+  TOLERATE_MISSING_SCHEMA("orc.tolerate.missing.schema",
+      "hive.exec.orc.tolerate.missing.schema",
+      true,
+      "Writers earlier than HIVE-4243 may have inaccurate schema metadata.\n"
+          + "This setting will enable best effort schema evolution rather\n"
```

Comment:
I'd suggest that the comment include something about using position in the top level structure as the fallback and that it support adding new columns to the end and changing types, but not deleting or reordering the top level columns.